tests/test_index.cpp

```diff
@@ -254,11 +254,53 @@ TestIndex::TestSerializeFile(const IndexPtr& index_from,
         }
     }
 }
+
+void
+TestIndex::TestConcurrentAdd(const TestIndex::IndexPtr& index,
+                             const TestDatasetPtr& dataset,
+                             bool expected_success) {
+    auto base_count = dataset->base_->GetNumElements();
+    int64_t temp_count = base_count / 2;
+    auto dim = dataset->base_->GetDim();
+    auto temp_dataset = vsag::Dataset::Make();
+    temp_dataset->Dim(dim)
+        ->Ids(dataset->base_->GetIds())
+        ->NumElements(temp_count)
+        ->Float32Vectors(dataset->base_->GetFloat32Vectors())
+        ->Owner(false);
+    index->Build(temp_dataset);
+    auto rest_count = base_count - temp_count;
+    fixtures::ThreadPool pool(5);
+    using RetType = tl::expected<std::vector<int64_t>, vsag::Error>;
+    std::vector<std::future<RetType>> futures;
+
+    auto func = [&](uint64_t i) -> RetType {
+        auto data_one = vsag::Dataset::Make();
+        data_one->Dim(dim)
+            ->Ids(dataset->base_->GetIds() + i)
+            ->NumElements(1)
+            ->Float32Vectors(dataset->base_->GetFloat32Vectors() + i * dim)
+            ->Owner(false);
+        auto add_index = index->Add(data_one);
+        return add_index;
+    };
+
+    for (uint64_t j = rest_count; j < base_count; ++j) {
+        futures.emplace_back(pool.enqueue(func, j));
```

Comment:
 `pool` is not a task pool, so should multiple batches of write tasks be submitted here?